# Discrete_Time_Markov_Chain_Evolution
 Simulating phenotype evolution using discrete time Markov chains
 
 
 ![Picture1](https://user-images.githubusercontent.com/6492012/138521422-581c6d6f-586a-4a81-80a8-369b502e1b56.png)


The principles behind our simulation model: A. We simulated the evolution of fluorescence intensity of GFP populations. Each cell contains GFP molecules whose fluorescence output and folding stability can vary in a population by mutations. The GFP abundance also varies from cell to cell (gene expression noise) but does not evolve as the promoters are not mutated during the experiments. Evolution proceeds by two phases of mutation and selection. The details of these phases are shown in blue and green boxes, respectively. We partitioned the range of sensible phenotypic values for both folding stability and fluorescence output into n=100 equidistant bins, ranging from the lowest value of the property of interest to its highest value. We refer to these bins as states in our model. B. We sampled a distribution of the stability of single point mutations calculated by the free energy estimator, FoldX. In the model, we assume that the stability of a pre-mutated protein (shown as WT), can change to different values with different transition probabilities (Equation S13). Calculating these transition probabilities for all possible stability values give us a transition probability matrix. C) We sampled an empirically known distribution of the fluorescence intensity of a library of single point mutations to generate GFP variants with different fluorescence outputs (Equation S12). Similar to the case of protein stability, the fluorescence output of a pre-mutated protein can change to different values based on their transition probabilities. D. Collectively, mutations in folding stability and fluorescence output generate variants with different values of these properties in the population. E. To simulate the mutation process more efficiently, we use a formalism akin to Markov chains and consider 10000 trajectories from the initial states of WT protein. We advance these 2000 chains for one step and record these states which gives rise to 10000 values for protein stability and fluorescence output. F. The role of protein abundance is to increase the fluorescence intensity of our simulated cells. However and because of gene expression noise, some cells will have a higher fluorescence intensity than others. G. We implemented selection by letting a fixed percentile of cells with high fluorescence survive. Stronger selection leads to the survival of molecules whose fluorescence output and stability is higher. H. Upon selection we create a new WT protein whose fluorescence output and stability is the average of these properties for selected GFPs. This WT protein is then ready to start 10000 new simulation trajectories and continue the cycles of mutation and selection. 


